app/screens/home/StoreList.java

```diff
@@ -1,26 +1,68 @@
 package icebase.app.screens.home;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Scanner;
 import icebase.app.App;
+import icebase.app.api.API;
+import icebase.app.enums.SCREEN_ENUM;
 import icebase.app.MenuTitle;
+import icebase.app.Router;
 import icebase.app.screens.Screen;
+import icebase.app.types.Store;
+import icebase.icebase.Auth;
+import icebase.icebase.User;
 
 public class StoreList implements Screen {
     public void display() {
         Scanner sc = App.sc;
-
-        MenuTitle.displayMainTitle();
-        MenuTitle.displaySubTitle("[STORES]");
-        String[] storeList = { "Store 1", "Store 2" };
-        int range = storeList.length;
-        storeList[range - 1] = "Return";
         int choice;
+        Store chosenStore;
+        Auth currentUser = Auth.getAuth();
+
+        List<Store> stores = new ArrayList<>();
+        ArrayList<String> storeNames = new ArrayList<>();
+
+        // Gets names of each stores for options list
+        try {
+            stores = API.getStores();
+
+            for (Store store : stores) {
+                storeNames.add(store.getName());
+            }
+            storeNames.add("Return");
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            return;
+        }
+
+        while (true) {
+            MenuTitle.displayMainTitle();
+            MenuTitle.displaySubTitle("STORES");
+            MenuTitle.displayOptions(storeNames);
+
+            try {
+                System.out.print("Choice: ");
+                choice = Integer.parseInt(sc.nextLine());
+                if (choice == storeNames.size()) {
+                    return;
+                }
+
+                chosenStore = stores.get(choice - 1);
+                if (chosenStore.getSellerId().equals(currentUser.getUser().getId())) {
+                    Router.navigate(SCREEN_ENUM.SELLING_VIEW);
+                } else {
+                    // Router.navigate(SCREEN_ENUM.BUYING_VIEW);
+
+                    // For testing
+                    System.out.println("BUYING VIEW");
+                }
+            } catch (Exception e) {
+                System.out.println("Please choose from the given options...\n");
```

Comment:
hmmm may other exceptions pa ta nga need i consider and not only if out of bounds ang exception,

you can still have this print statement pero specify which exception this may happen - probably the out of bounds nga exception

ang general exception better nga i print out ang `e.getMessage()`